Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -557,60 +557,100 @@ def fetch_resource_and_forced_version(formula_or_resource, new_version, url, **s
-        formula.resources.select { |r| r.livecheck.formula == :parent }.each do |resource|
-          new_url = update_url(resource.url, resource.version.to_s, version)
-
-          if new_url == resource.url
-            opoo <<~EOS
-              You need to bump resource "#{resource.name}" manually since the new URL
-              and old URL are both:
-                #{new_url}
-            EOS
-            next
+      def update_livecheck_defined_resources!(formula, formula_version:)
+        formula.resources.each do |resource|
+          next unless resource.livecheck_defined?
+          next if resource.livecheck.skip?
+
+          old_url = resource.url
+          old_revision = resource.specs[:revision]
+          old_version = resource.version.to_s
+          new_version = formula_version
+
+          if resource.livecheck.formula != :parent
+            require "livecheck/livecheck"
+            version_info = Homebrew::Livecheck.resource_version(resource, formula_version,
+                                                                json:    true,
+                                                                debug:   args.debug?,
+                                                                quiet:   args.quiet?,
+                                                                verbose: args.verbose?)
+            new_version = version_info.dig(:version, :latest)
+            odie "Resource \"#{resource.name}\" livecheck failed to get latest version!" if new_version.blank?
+            next if old_version == new_version
           end
 
-          new_mirrors = resource.mirrors.map do |mirror|
-            update_url(mirror, resource.version.to_s, version)
+          if old_revision.present?
+            new_revision = if old_revision == old_version
+              forced_version = true
+              new_version
+            elsif (old_tag = resource.specs[:tag])
+              new_tag = old_tag.gsub(old_version, new_version)
+              odie "Resource \"#{resource.name}\" tag could not be updated!" if old_tag == new_tag
+              resource_path, forced_version = fetch_resource_and_forced_version(resource, new_version, old_url,
+                                                                                tag: new_tag)
+              Utils.popen_read("git", "-C", resource_path.to_s, "rev-parse", "-q", "--verify", "HEAD").strip
+            else
+              odie <<~EOS
+                Unable to automatically bump resource "#{resource.name}"!
+                Either bump this formula manually or add support to `bump-formula-pr`.
+              EOS
+            end
+            # Allow skipping different tags on same revision
+            next if old_revision == new_revision
+
+            inreplace_regex = /
+              ^(?<ws>[ ]+)resource\ "#{resource.name}"\ do\s+
+                url\ "#{Regexp.escape(old_url)}",\s+
+                    (using:[ ]+.*,\s+)?
+                    (tag:[ ]+.*,\s+)?
```

Comment:
```suggestion
              ^(?<ws> +)resource\ "#{resource.name}"\ do\s+
                url\ "#{Regexp.escape(old_url)}",\s+
                    (using: +.*,\s+)?
                    (tag: +.*,\s+)?
```

Unless I'm overlooking something, I believe you should be able to use ` +` here instead of `[ ]+`. The previous regex was using `[ ]+` because the space was at the very start of the line but that's not the case here.